tools/xabuild/XABuildPaths.cs

```diff
@@ -161,7 +161,9 @@ public XABuildPaths ()
 			}
 
 			FrameworksDirectory       = Path.Combine (prefix, "xbuild-frameworks");
-			MSBuildExtensionsPath     = Path.Combine (prefix, "xbuild");
+
+			MSBuildExtensionsPath     = Path.Combine (IsMacOS ? "/Library/Frameworks/Mono.framework/Versions/Current/lib/mono" : "/usr/lib/mono", "xbuild");
```

Comment:
This is replacing our "in-tree" location with a *system-global* location, thus defeating the ~entire purpose of having `xabuild` in the first place.

That seems bad. :-)

Reply:
How? The order of the lookup for imports will be:

1. default/global location (which won't have XA available, because that would be in `External` for that system mono)
2. in-tree location (which should have XA)
3. `External` from system mono

.. due to the change `ProjectImportSearchPaths = new [] { Path.Combine (prefix, "xbuild"), Path.Combine (monoExternal, "xbuild") };` .

This should mean that xabuild will cause projects to be built against the in-tree XA and not system XA. IIUC, that is what you need but please correct me if I'm mistaken.

Reply:
@radical can you a attach a quick binlog with this change, and I'll take a look:
```
./bin/Debug/bin/xabuild samples/HelloWorld /bl
```

I think it _might_ work (especially since this PR is green), and perhaps this ordering trickled over from Windows. I did most of the development of `xabuild.exe` on Windows.

Reply:
AFAICS, this is the same ordering w.r.t XA targets/tasks, because existing code has (1) and (2) swapped. But XA is only available in (2) and (3), so their relative ordering remains the same.

I don't have a local build of XA! If the above comment does not satisfy then I can contact you over slack,  build XA locally and send you the binlog.

Reply:
Seems to be OK, here is a binlog I produced on Mac (opened on Windows).

Right targets are imported:
![image](https://user-images.githubusercontent.com/840039/49461609-ab50a680-f7b9-11e8-884b-3faadc3959df.png)

Tasks are loaded from the correct assembly:
![image](https://user-images.githubusercontent.com/840039/49461639-bf94a380-f7b9-11e8-8427-7f90967307d9.png)

Reply:
@radical wrote:

> How? The order of the lookup for imports will be:
> 
> 1. default/global location (which won't have XA available, because that would be in External for that system mono)
> 2. in-tree location (which should have XA)
> 3. External from system mono

Which is wrong: (2) should come *before* (1), so that when I use the build on my local (non-CI) machine -- which has the global location installed! -- I use the in-tree installation and *not* the system installation, because Parallel Installs Are Useful.

Reply:
> `so that when I use the build on my local (non-CI) machine -- which has the global location installed! -- I use the in-tree installation and not the system installation`

And it will. Because the global location means `/Library/Frameworks/Mono.frameworks/Versions/Current/lib/mono/xbuild`, and that does not contain XA. XA is found in `/Library/Frameworks/Mono.frameworks/External/xbuild` which is (3).
So, the order is:

1. `/Library/Frameworks/Mono.frameworks/Versions/Current/lib/mono/xbuild` (No XA)
2. in-tree location (which should have XA)
3. `/Library/Frameworks/Mono.frameworks/External/xbuild` (System XA)

Reply:
I hadn't properly differentiated between (1) and (3).

Another problem here is *Windows support*: the paths specified won't work on Windows.

Thus, two possible fixes/questions:

 1. Is `MSBuildExtensionsPath` needed anymore?  If we altered `xabuild` to *not* override `MSBuildExtensionsPath` -- and instead use the MSBuild default values -- what would happen?
 2. Or, should this use `typeof(object).Assembly.Location` or something?

I'm wondering if (1) is the better path...